RELEASE_NOTES.rst

```diff
@@ -21,6 +21,276 @@
 
 .. towncrier release notes start
 
+Airflow 2.9.0 (2024-03-28)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+
+Following Listener API methods are considered stable and can be used for production system (were experimental feature in older Airflow versions) (#36376):
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+Lifecycle events:
+
+- ``on_starting``
+- ``before_stopping``
+
+DagRun State Change Events:
+
+- ``on_dag_run_running``
+- ``on_dag_run_success``
+- ``on_dag_run_failed``
+
+TaskInstance State Change Events:
+
+- ``on_task_instance_running``
+- ``on_task_instance_success``
+- ``on_task_instance_failed``
```

Comment:
Actually, this is an opportunity to change on_task_instnce_failed to contain error message. I have not realized those were experimental, but that would solve the problem that @vandonr pointed at in https://github.com/apache/airflow/pull/38155  (maybe?)

Reply:
commented in #38155. I will go ahead with the alpha and cherry-pick when necessary.

Reply:
Ah, so should I go ahead and change the signature before they are marked as good to go in this release ?
Your comment @ephraimbuddy was about doing the change in a non-breaking way 🤔